### ISSUE TITLE
要望: inputsecret での状態バリアント

### ISSUE THREAD

**Issue by tsuyoshicho** [vim-jp/issues#1244]
## 質問・報告の内容
inputsecret()は常に

1. ※ 表示
2. 履歴にのこさない

ですが、1については、最近の事情を鑑みると

- ※ 表示
- 最後の1文字は見える
- ※ と 「全部見える」をなにかでトグル
- (insecure) 全部見える、けど履歴に残らない

の引数とかオプションでの切り替えができると嬉しそうです。

トグルは難しいでしょうけども...GUIだけ特殊キーで、くらいはあってもいいかも?

## バージョン他
VIM - Vi IMproved 8.1 (2018 May 18, compiled Mar  2 2019 15:46:47)
MS-Windows 64 ビット コンソール 版
適用済パッチ: 1-990

Windows 10 Pro 64bit

**Comment by ichizok** [comment]
https://gist.github.com/ichizok/7c5b6cfadea5018bf329980da0db14c8
簡易的に実装してみました。Ctrl-X でトグル。
最後の1文字も一定時間経過後にマスクされるのが望ましいと思いますが、難しいかも。

**Comment by tsuyoshicho** [comment]
でしょうねえ...
最後の1文字は、そのまんま系と、時間でマスクされる系があるようですし、できてもそのまんまでもいいかなという気がします...

**Comment by ichizok** [comment]
投げました。
https://github.com/vim/vim/pull/4692

**Comment by ichizok** [comment]
https://github.com/vim/vim/pull/4692#issuecomment-512781490
否定的な反応。

ディスプレイへのショルダーハッキング防御でしかないので、ユーザビリティを犠牲にするほどのものか（表示切り替えくらいあってもいいのではないか）という気はします。

入力をマスクすることはセキュリティに貢献しないという論説（主にWebの入力フォームの話ですが）
https://u-site.jp/alertbox/20090623_passwords
https://www.nngroup.com/articles/stop-password-masking/

**Comment by tsuyoshicho** [comment]
はい、昨今の傾向として、ショルダーハックは阻止できるようにしつつ、視認できるようにしてパスワードの複雑度向上を補助する感じですよね。
すくなくとも最後の1文字は見たい

\# なのでこのIssueでした

---

自分が最初に見掛けた資料は...あった
https://www.slideshare.net/ockeghem/uxvssec20170609

![page 12](https://image.slidesharecdn.com/uxvssec20170609-170609123003/95/-12-638.jpg)

他に
https://blog.tokumaru.org/2012/06/cookpad-display-password.html
など

**Comment by k-takata** [comment]
vim/vim#4692 はrejectされましたね。

**Comment by tsuyoshicho** [comment]
むうん、しょうがないかな
